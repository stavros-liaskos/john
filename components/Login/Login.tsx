import React from 'react';
import { LoginProps } from './Login.types';

const Login: React.FunctionComponent<LoginProps> = ({ i18n, handleRegister }) => {
  if (!i18n?.loginBtn || !i18n.text || !i18n.welcome || !i18n.artistsCount || !i18n.releasesCount) {
    return null;
  }

  return (
    <div className="flex flex-col items-center justify-center lg:w-1/2">
      <h2 className="rr-text text-4xl mb-3">{i18n.welcome}</h2>
      <p className="rr-text text-xl mb-3">{i18n.text}</p>

      <button className="btn btn-large lg:ml-8 my-8 w-44 h-14" onClick={() => handleRegister()}>
        {i18n.loginBtn}
      </button>
    </div>
  );
};

export default Login;
