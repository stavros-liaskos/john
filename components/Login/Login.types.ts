type LoginI18n = {
  welcome: string;
  loginBtn: string;
  text: string;
  artistsCount: string;
  releasesCount: string;
};

export interface LoginProps {
  i18n: LoginI18n;
  handleRegister: Function;
}
