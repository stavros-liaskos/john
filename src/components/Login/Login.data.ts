export const loginI18n = {
  welcome: 'Welcome to Release Raccoon!',
  loginBtn: 'Register',
  text: "Receive your favorite artists' music in your email every week!",
  artistsCount: 'Artists',
  releasesCount: 'Releases',
};
