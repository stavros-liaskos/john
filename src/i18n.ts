export const metaI18n = {
  title: 'Release Raccoon',
  description: 'A music release newsletter',
  url: 'https://www.releaseraccoon.online',
};
