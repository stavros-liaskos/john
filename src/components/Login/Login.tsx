import React from 'react';
import { LoginProps } from './Login.types';
import Link from 'next/link';

const Login: React.FunctionComponent<LoginProps> = ({ i18n }) => {
  if (!i18n?.loginBtn || !i18n.text || !i18n.welcome || !i18n.artistsCount || !i18n.releasesCount) {
    return null;
  }

  return (
    <div className="flex flex-auto flex-col items-center justify-center w-full">
      <h2 className="rr-text text-center text-4xl mb-3">{i18n.welcome}</h2>
      <p className="rr-text text-center text-xl mb-3">{i18n.text}</p>

      <Link href="https://api.releaseraccoon.online/me?redirectUrl=https://www.releaseraccoon.online">
        <button className="btn btn-large lg:ml-8 mt-8 mb-24 w-44 h-14">{i18n.loginBtn}</button>
      </Link>

      <Link href="/api/auth/login">
        <button className="btn btn-large lg:ml-8 mt-8 mb-24 w-44 h-14">2. Take me to dashboard</button>
      </Link>
    </div>
  );
};
Login.whyDidYouRender = true;
export default Login;
