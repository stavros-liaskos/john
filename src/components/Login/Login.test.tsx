import React from 'react';
import Login from './Login';
import { render } from '@testing-library/react';
import { loginI18n } from '../../i18n';

describe('Login', () => {
  it('renders without data without crashing', () => {
    // @ts-ignore
    render(<Login />);
  });

  it('renders login button', () => {
    const { container } = render(<Login i18n={loginI18n} />);

    expect(container).toMatchSnapshot();
  });
});
