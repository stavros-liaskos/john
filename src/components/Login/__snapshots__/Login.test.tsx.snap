// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders login button 1`] = `
<div>
  <div
    class="flex flex-auto flex-col items-center justify-center w-full"
  >
    <h2
      class="rr-text text-center text-4xl mb-3"
    >
      Welcome to Release Raccoon!
    </h2>
    <p
      class="rr-text text-center text-xl mb-3"
    >
      Receive your favorite artists' music in your email every week!
    </p>
    <a
      href="/api/auth/login"
    >
      <button
        class="btn btn-large lg:ml-8 mt-8 mb-24 w-44 h-14"
      >
        Register
      </button>
    </a>
  </div>
</div>
`;
